src/shared/identityUtils.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+
+import fs from 'node:fs';
+import { randomBytes } from 'node:crypto';
+import { DevServerUtils } from './devServerUtils.js';
+
+class LwrConfigFile {
+  public identityToken?: string;
+}
+
+const BYTES_TO_GENERATE = 32;
+
+export class IdentityUtils {
+  public static async createIdentityToken(): Promise<void> {
+    const lwrConfigFile = DevServerUtils.getServerConfigFileLocation();
+    if (fs.existsSync(lwrConfigFile)) {
```

Comment:
Isn't it necessary to have the identity stored in the config file otherwise preview will not work? If that is the case then the only time this method should resolved would be when it successfully updates the config file. All other cases should throw.

@khawkins is my understanding correct here?

Reply:
The extension to that argument, I would argue, is that if an identity token already exists, then, do nothing(resolve) as it already satisfies the condition of needing to have an identity token. Maybe the method name of `createIdentityToken()` isn't perfect as it implies that it needs to create a token every time it is getting called.

Reply:
Oh yes I get that part. In fact the suggested method implementation that I've provided [here](https://github.com/salesforcecli/plugin-lightning-dev/pull/52#discussion_r1633931128) accounts for the case where `identityToken` already exists but with less if-statements and logic.